### PR TITLE
Use sync errors with more information.

### PR DESCRIPTION
**Pull request by nickmshelley**

### CODE REVIEW

File: LDSAnnotations/SyncAnnotationsOperation.swift [hunk 1]
```diff
@@ -82,14 +78,21 @@ class SyncAnnotationsOperation: Procedure, ResultInjection {
...
+                }
+                completion(.error(errors: syncErrors))
             }
         }))
     }
     
     override func execute() {
         guard requirement.value != nil else {
-            finish(withError: SyncAnnotationsOperationError.notebookSyncFailed("Sync notebooks failed, cannot sync annotations."))
+            finish(withError:SyncError(id: nil, username: session.username, message: "Sync notebooks failed, cannot sync annotations.", json: nil, type: .unknown))
```
Comment (stephanheilner) [inline LDSAnnotations/SyncAnnotationsOperation.swift]: `(withError: SyncError`

**Comment by nickmshelley** [comment]
Tests pass locally

File: LDSAnnotations/SyncAnnotationsOperation.swift [hunk 2]
```diff
@@ -82,14 +78,21 @@ class SyncAnnotationsOperation: Procedure, ResultInjection {
...
             } else {
-                completion(.error(errors: errors))
+                let syncErrors: [SyncError] = errors.map { error in
+                    if let syncError = error as? SyncError {
+                        return syncError
+                    } else {
+                        return SyncError(id: nil, username: session.username, message: "\(error)", json: nil, type: .unknown)
+                    }
+                }
+                completion(.error(errors: syncErrors))
```
Comment (stephanheilner) [inline LDSAnnotations/SyncAnnotationsOperation.swift]: This seems like it will still mark sync as a failure and not get the changes that come as well. Or am I reading this wrong?

Comment (nickmshelley) [inline LDSAnnotations/SyncAnnotationsOperation.swift]: Yeah I decided to do that as a separate commit because they aren't really related and I was already well into this change.

**Comment by nickmshelley** [comment]
I didn't rerun tests because I just added a space.

**Comment by stephanheilner** [comment]
👍
